gqcp/src/ONVBasis/ONVPath.cpp

```diff
@@ -0,0 +1,164 @@
+    this->m_address -= this->onv_basis.arcWeight(q, n);
+
+    // Update the next possible creation index. Since we're always constructing paths from the top-left to the bottom-right, we're only considering creation indices p > q.
+    this->orbital_index = q + 1;
+    this->electron_index = n;
+}
+
+/**
+ *  Create the diagonal arc that starts at the coordinate (p, n).
+ * 
+ *  @param p        the index of the orbital that should be created
+ *  @param n        the number of electrons in the ONV/path up to the orbital index q, prior to the creation
+ */
+void ONVPath::create() {
+
+    // During creation, we're adding an arc, so we'll have to update the current address by adding the corresponding arc weight.
+    this->m_address += this->onv_basis.arcWeight(this->orbital_index, this->electron_index);
+
+    // Update the next possible creation index. Since creation means adding a diagonal, we must update both orbital_index and electron_index.
+    this->orbital_index++;
+    this->electron_index++;
+}
+
+/**
+ *  Create the diagonal arc that starts at the coordinate (p, n).
+ * 
+ *  @param p        the index of the orbital that should be created
+ *  @param n        the number of electrons in the ONV/path up to the orbital index q, prior to the creation
+ */
+void ONVPath::create(const size_t p, const size_t n) {
+
+    // During creation, we're adding an arc, so we'll have to update the current address by adding the corresponding arc weight.
+    this->m_address += this->onv_basis.arcWeight(p, n);
+
+    // Update the next possible creation index. Since creation means adding a diagonal, orbital_index > p and electron_index > n.
+    this->orbital_index = p + 1;
+    this->electron_index = n + 1;
+}
+
+/**
+ * Return if the path has been finished, i.e. if it the electron index has exceeded the total number of electrons, keeping in mind that we start at index 0.
+ * 
+ * @return if the path has been finished
+ */
+ bool ONVPath::isFinished() {
+    return this->electron_index >= this->onv_basis.numberOfElectrons();
+ }
+
+/**
+ *  Translate the diagonal arc that starts at the coordinate (p, n) to the left.
+ * 
+ *  @param p        the index of the orbital that should be annihilated
+ *  @param n        the number of electrons in the ONV/path up to the orbital index p
+ */
+void ONVPath::leftTranslate(const size_t p, const size_t n) {
+
+    // Translating a diagonal arc can be rewritten as an removal of the current arcweight, followed by an addition of the arcweight on the left.
+    this->m_address += this->onv_basis.arcWeight(p, n-1) - this->onv_basis.arcWeight(p, n);
+
+    // Since a left-translation describes the process of 'encountering an electron/occupied orbital', the sign factor should be updated according to the fermionic anticommutation rules.
+    this->m_sign *= -1;
+
+    // Update the internal orbital and electron indices to move to the next orbital.
+    this->orbital_index++;
+    this->electron_index++;
```

Comment:
This seems wrong in a general context, since there isn't any reference to the given indices `p` or `n`.